src/groupByKey.ts

```diff
@@ -2,6 +2,15 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  return items.reduce((result, item) => {
+    const keyValue = item[key] as unknown as string;
```

Comment:
Casting to `unknown` and then to `string` can be unsafe if `item[key]` is not actually a string. This can lead to unexpected behavior if `item[key]` is an object or other non-string data type. Ensure that the key provided will always result in a string value, or handle the case where it does not.